### PR TITLE
Benchmarks: Add Feature - Add _post_process() interface.

### PR DESCRIPTION
**Pull request by guoshzhao**
Major changes:
- add _post_process() interface in base.py.
- define new return code: DISTRIBUTED_SETTING_DESTROY_FAILURE.

### CODE REVIEW

File: superbench/benchmarks/micro_benchmarks/sharding_matmul.py [hunk 1]
```diff
@@ -221,7 +217,7 @@ def _benchmark(self):
         K = self._args.k
         N = self._args.n
         for mode in self._args.mode:
-            if mode == ShardingMode.NOSHARDING or self.__world_size == 1:
+            if mode == ShardingMode.NOSHARDING:
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/sharding_matmul.py:220]: Why remove self.__world_size == 1?

File: superbench/benchmarks/return_code.py [hunk 2]
```diff
@@ -18,7 +18,8 @@ class ReturnCode(Enum):
     MODEL_TRAIN_FAILURE = 11
     MODEL_INFERENCE_FAILURE = 12
     DISTRIBUTED_SETTING_INIT_FAILURE = 13
-    DATASET_GENERATION_FAILURE = 14
-    DATALOADER_INIT_FAILURE = 15
-    OPTIMIZER_CREATION_FAILURE = 16
-    MODEL_CREATION_FAILURE = 17
+    DISTRIBUTED_SETTING_DESTROY_FAILURE = 14
```
Comment (cp5555) [inline superbench/benchmarks/return_code.py:21]: Why not add it as 18 directly?

File: superbench/benchmarks/micro_benchmarks/sharding_matmul.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 import os
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/sharding_matmul.py:14]: Shall we separate it in another PR? It's not the interface actually.

File: superbench/benchmarks/micro_benchmarks/sharding_matmul.py (hunk 1 above)
Comment (guoshzhao) [reply superbench/benchmarks/micro_benchmarks/sharding_matmul.py:220]: The reason is introduced in the description of PR:

> For self.__matmul_nosharding() change the condition from 'if mode == ShardingMode.NOSHARDING or self.__world_size == 1:' to 'if mode == ShardingMode.NOSHARDING', since sharding-matmul with 1GPU should also be OK to run.

File: superbench/benchmarks/return_code.py (hunk 2 above)
Comment (guoshzhao) [reply superbench/benchmarks/return_code.py:21]: Just put it near the DISTRIBUTED_SETTING_INIT_FAILURE error.

File: superbench/benchmarks/micro_benchmarks/sharding_matmul.py (hunk 3 above)
Comment (guoshzhao) [reply superbench/benchmarks/micro_benchmarks/sharding_matmul.py]: Done.

**Review by cp5555 — APPROVED** [review]

**Review by abuccts — APPROVED** [review]
